src/EFCore.SqlServer/Extensions/SqlServerServiceCollectionExtensions.cs

```diff
@@ -30,6 +31,81 @@ namespace Microsoft.Extensions.DependencyInjection
     /// </summary>
     public static class SqlServerServiceCollectionExtensions
     {
+        /// <summary>
+        ///     <para>
+        ///         Registers the given Entity Framework context as a service in the <see cref="IServiceCollection" />
+        ///         and configures it to connect to a SQL Server database.
+        ///     </para>
+        ///     <para>
+        ///         Use this method when using dependency injection in your application, such as with ASP.NET Core.
+        ///         For applications that don't use dependency injection, consider creating <see cref="DbContext" />
+        ///         instances directly with its constructor. The <see cref="DbContext.OnConfiguring" /> method can then be
+        ///         overridden to configure the SQL Server provider and connection string.
+        ///     </para>
+        ///     <para>
+        ///         To configure the <see cref="DbContextOptions{TContext}" /> for the context, either override the
+        ///         <see cref="DbContext.OnConfiguring" /> method in your derived context, or use the appropriate
+        ///         <see cref="EntityFrameworkServiceCollectionExtensions.AddDbContext{TContext}(IServiceCollection, Action{DbContextOptionsBuilder}?, ServiceLifetime, ServiceLifetime)"/>
+        ///         method and supply an optional action to configure the <see cref="DbContextOptions" /> for the context. 
+        ///     </para>
+        ///     <para>
+        ///         For more information on how to use this method, see the Entity Framework Core documentation at https://aka.ms/efdocs.
+        ///         For more information on using dependency injection, see https://go.microsoft.com/fwlink/?LinkId=526890.
+        ///     </para>
+        /// </summary>
+        /// <typeparam name="TContext"> The type of context to be registered. </typeparam>
+        /// <param name="serviceCollection"> The <see cref="IServiceCollection" /> to add services to. </param>
+        /// <param name="connectionString"> The connection string of the database to connect to. </param>
+        /// <param name="sqlServerOptionsAction"> An optional action to allow additional SQL Server specific configuration. </param>
+        /// <returns> The same service collection so that multiple calls can be chained. </returns>
+        public static IServiceCollection AddSqlServer<TContext>(this IServiceCollection serviceCollection, string connectionString, Action<SqlServerDbContextOptionsBuilder>? sqlServerOptionsAction = null)
```

Comment:
Alternative proposal:
```suggestion
        public static IServiceCollection AddSqlServerDbContext<TContext>(this IServiceCollection serviceCollection, string connectionString, Action<SqlServerDbContextOptionsBuilder>? sqlServerOptionsAction = null, Action<IServiceProvider, DbContextOptionsBuilder>? optionsAction = null)
```

Reply:
I'm not sure I understand the suggestion here. Does it make sense to allow both `sqlServerOptionsAction` and `optionsAction` to be specified? If we did allow it, is the ordering intuitive?

Reply:
Looks like the underlying AddDbContext has separate overloads that take in a `Action<IServiceProvider, DbContextOptionsBuilder>?` or `Action<DbContextOptionsBuilder>?` but not both. Should we follow that pattern here?

Reply:
Since this is meant to be simpler we can have just the `Action<DbContextOptionsBuilder>` overload.

@ajcvickers cc

Reply:
> Does it make sense to allow both sqlServerOptionsAction and optionsAction to be specified? If we did allow it, is the ordering intuitive?

Yes, both or any can be specified. Since the method is about the provider it makes sense to put the provider options first.